arch/risc-v/include/arch.h

```diff
@@ -42,70 +30,47 @@
  ****************************************************************************/
 
 #include <nuttx/config.h>
-
-#ifndef __ASSEMBLY__
-#  include <stdint.h>
-#endif
-
 #include <arch/csr.h>
 
-#ifdef CONFIG_ARCH_RV32IM
-#  include <arch/rv32im/arch.h>
-#endif
-
-#ifdef CONFIG_ARCH_RV64GC
-#  include <arch/rv64gc/arch.h>
-#endif
-
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 
-/* Macros to get the core and vendor ID, HART, arch and ISA codes, etc.
- */
-
-#ifdef CONFIG_RV32IM_SYSTEM_CSRRS_SUPPORT
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
 
-uint32_t up_getmisa(void);
-uint32_t up_getarchid(void);
-uint32_t up_getimpid(void);
-uint32_t up_getvendorid(void);
-uint32_t up_gethartid(void);
+#ifndef __ASSEMBLY__
 
+#ifdef __LP64__
+typedef uint64_t reg_t;
```

Comment:
@xiaoxiang781216 How do you think we should approach this?  I would prefer to use something like uintptr_t, but that does not feel quite right either.

Reply:
uintptr_t looks fine at least for riscv_getsp